example/write-struct/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/stellentus/go-plc"
+)
+
+var (
+	addr     = flag.String("address", "192.168.1.176", "Hostname or IP address of the PLC")
+	plcDebug = flag.Int("plctagdebug", 0, "Debug level for libplctag's debug (0-5)")
+)
+
+type DemoTags struct {
+	Speed    int64
+	Duration float32 `plctag:"THE_DURATION"`
+	Level    int32   `plctag:",omitempty"`
+}
+
+// This command demonstrates writing a struct.
+// It will write the struct, then read each value individually.
+func main() {
+	flag.Parse()
+
+	plc.SetLibplctagDebug(plc.LibplctagDebugLevel(*plcDebug))
+
+	fmt.Printf("Initializing connection to %s\n", *addr)
+
+	device, err := plc.NewDevice(*addr)
+	panicIfError(err, "Could not create test PLC!")
+	defer func() {
+		err := device.Close()
+		panicIfError(err, "Close was unsuccessful")
+	}()
+
+	// First write the Level to show it's not overwritten later
+	originalLevel := int32(13)
+	err = device.WriteTag("DUMMY_STRUCT.Level", originalLevel)
+	panicIfError(err, "Couldn't write Level")
+
+	demoVal := DemoTags{
+		Speed:    7738,
+		Duration: 3.14,
+		Level:    0, // Since this is 0 and is 'omitempty', it won't overwrite the existing value
+	}
+
+	// Wrap the device in a SplitWriter to allow a struct to be written
+	err = plc.NewSplitWriter(device).WriteTag("DUMMY_STRUCT", demoVal)
+	panicIfError(err, "Couldn't read dummy struct")
+
+	// Demonstrate reading each of the three struct values independently
+	var speed int64
+	err = device.ReadTag("DUMMY_STRUCT.Speed", &speed)
+	panicIfError(err, "Couldn't read Speed")
+	fmt.Printf("Read speed as %d (expecting %d)\n", speed, demoVal.Speed)
```

Comment:
Since we panic on a read failure, should we perhaps also panic if we don't read the expected value?

Reply:
I wanted to avoid checks like that in an example. (I think a lot of example code would just ignore the error entirely, but I think that is bad practice in case a lazy dev just copy-pastes the example code and doesn't realize there's an error to check.)

Reply:
Ah, fair enough, given it's more an example and not a proper test.